reframe/core/fields.py

```diff
@@ -105,23 +107,32 @@ def __set__(self, obj, value):
 
 
 class TimerField(TypedField):
-    '''Stores a timer in the form of a tuple ``(hh, mm, ss)``'''
+    '''Stores a timer in the form of a :class:`datetime.timedelta` object'''
 
     def __init__(self, fieldname, *other_types):
-        super().__init__(fieldname, types.Tuple[int, int, int], *other_types)
+        super().__init__(fieldname, datetime.timedelta, str,
+                         types.Tuple[int, int, int], *other_types)
 
     def __set__(self, obj, value):
         self._check_type(value)
-        if value is not None:
-            # Check also the values for minutes and seconds
+        if isinstance(value, tuple):
+            user_deprecation_warning(
+                "setting a timer field from a tuple is deprecated: "
+                "please use a string <days>d<hours>h<minutes>m<seconds>s")
```

Comment:
Better use the single quotes for the string

Reply:
Sorry, my bad, I just saw @vkarak told you to change it. Just add the single quotes around `<days>d<hours>h<minutes>m<seconds>s`.

Reply:
Yes, please add the single quotes around the format. That's why I asked for double quotes in the first place.